### PR TITLE
add precourse

### PR DESCRIPTION
**Pull request by maz-e**

### CODE REVIEW

File: staff/miguel-angel-zapatero/precourse/airlines-pro/index.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        return 'No hay vuelos!' 
+    }
+}
+
+function showAvgCost(data) {
+    let sum = 0;
+    data.forEach(function(flight){
+        sum += flight.cost; 
+    });
+    return 'El coste medio de los vuelos es ' + (sum / data.length).toFixed(2) + '€';
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/airlines-pro/index.js]: @maz-e as this code is right, you can improve this code by storing the average in a variable:

```
const average = (sum / data.length).toFixed(2);
return `El coste medio de los vuelos es ${average} €`;

```

File: staff/miguel-angel-zapatero/precourse/airlines/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+    return allFlights;
+}
+
+function showAvgCost(data) {
+    let sum = 0;
+    data.forEach(function(flight){
+        sum += flight.cost; 
+    }, sum);
+    return 'El coste medio de los vuelos es ' + (sum / data.length).toFixed(2) + '€';
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/airlines/index.js]: @maz-e Same here as above, this code is right, but you can store the operation into a variable and then print it in the return :)

File: staff/miguel-angel-zapatero/precourse/bingo-pro/index.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    
+    return arr.some(function(elem){
+        return elem === 5;
+    });
+}
+
+// Función para comprobar si el cartón está completo.
+function checkBingo(card) {
+    return card.some(function(elem){
+        return elem.mathed === false
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/bingo-pro/index.js]: @maz-e you left this return without ;

File: staff/miguel-angel-zapatero/precourse/bingo-pro/index.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    msg += '15 <= Every turn < 90 ---> 780 POINTS\n\n';
+    msg += 'LINE POINTS:\n';
+    msg += '5 <= Every turn < 88 ---> 500 POINTS\n\n'
+    msg += '0 POINTS no ranking!';
+    return msg;
+}
+
+// Función para crear ranking
+function makeRanking(name, num) {
+    return {userName: name, points: num}
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/bingo-pro/index.js]: @maz-e same here, miss ;

File: staff/miguel-angel-zapatero/precourse/bingo/index.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    
+    return arr.some(function(elem){
+        return elem === 5;
+    });
+}
+
+// Función para comprobar si el cartón está completo.
+function checkBingo(card) {
+    return card.some(function(elem){
+        return elem.mathed === false
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/bingo/index.js]: @maz-e same here, missing ;

File: staff/miguel-angel-zapatero/precourse/pasapalabra-2/index.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        strHTML += `<div class="rankitems"><p>${elem.falladas}</p></div>`;
+        document.getElementsByClassName('rankgrid')[0].insertAdjacentHTML("beforeend", strHTML);
+        i++;
+    });
+}
+
+function addRanking (name, good, bad) {
+    let points = (good * 10) - (bad * 5);
+    if (points > 0) {
+        return {playerName: name, puntos: points, correctas: good, falladas: bad}
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/pasapalabra-2/index.js]: @maz-e same here, missing ;

File: staff/miguel-angel-zapatero/precourse/bingo-pro/index.js (hunk 3 above)
Comment (carlosq09) [inline staff/miguel-angel-zapatero/precourse/bingo-pro/index.js]: `;` is missing

**Review by carlosq09 — CHANGES_REQUESTED** [review]

File: staff/miguel-angel-zapatero/precourse/pasapalabra-pro/index.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+function playAgain(data) {
+    if (confirm('Play again?')) {
+        return pasapalabra(data);
+    } else {
+        return 'Goodbye, see you again!';
+    }
+}
+
+function showResult(player, data){
+    return `${player}: ${goodAnswers(data)} correcta/s - ${badAnswers(data)} fallada/s`
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/pasapalabra-pro/index.js]: @maz-e missing ;

File: staff/miguel-angel-zapatero/precourse/pasapalabra/index.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    return data.filter(elem => elem.status == -1).length;
+}
+
+function passQuestions(data) {
+    return data.filter(elem => elem.status == 0);
+}
+
+// Función para crear ranking
+function makeRanking(name, num) {
+    return {playerName: name, points: num}
```
Comment (edfrias) [inline staff/miguel-angel-zapatero/precourse/pasapalabra/index.js]: @maz-e missing ;

File: staff/miguel-angel-zapatero/precourse/bingo-pro/index.js (hunk 3 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/bingo-pro/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/airlines-pro/index.js (hunk 1 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/airlines-pro/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/airlines/index.js (hunk 2 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/airlines/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/bingo-pro/index.js (hunk 3 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/bingo-pro/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/bingo-pro/index.js (hunk 4 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/bingo-pro/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/bingo/index.js (hunk 5 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/bingo/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/pasapalabra-2/index.js (hunk 6 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/pasapalabra-2/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/pasapalabra-pro/index.js (hunk 7 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/pasapalabra-pro/index.js]: 👍

File: staff/miguel-angel-zapatero/precourse/pasapalabra/index.js (hunk 8 above)
Comment (maz-e) [reply staff/miguel-angel-zapatero/precourse/pasapalabra/index.js]: 👍

**Pull request merged**